container/container.go

```diff
@@ -1,201 +1,520 @@
-	m := c.v2.GetPlacementPolicy()
-	if m == nil {
-		return nil
-	}
+// SetName sets human-friendly container domain name.
+func (x *Domain) SetName(name string) {
+	x.name = name
+}
 
-	var p netmap.PlacementPolicy
-	// FIXME(@cthulhu-rider): #225 handle error
-	err := p.ReadFromV2(*m)
-	if err != nil {
-		panic(err)
+// Name returns name set using SetName.
+//
+// Zero Domain has zero name.
+func (x Domain) Name() string {
+	return x.name
+}
+
+// SetZone sets zone which is used as a TLD of a domain name in NNS contract.
+func (x *Domain) SetZone(zone string) {
+	x.zone = zone
+}
+
+// Zone returns domain zone set using SetZone.
+//
+// Zero Domain has "container" zone.
+func (x Domain) Zone() string {
+	if x.zone != "" {
+		return x.zone
 	}
 
-	return &p
+	return "container"
 }
 
-func (c *Container) SetPlacementPolicy(v *netmap.PlacementPolicy) {
-	var m *v2netmap.PlacementPolicy
+// WriteDomain writes Domain into the Container. Name MUST NOT be empty.
+func WriteDomain(cnr *Container, domain Domain) {
+	cnr.SetAttribute(container.SysAttributeName, domain.Name())
+	cnr.SetAttribute(container.SysAttributeZone, domain.Zone())
+}
 
-	if v != nil {
-		m = new(v2netmap.PlacementPolicy)
-		v.WriteToV2(m)
+// ReadDomain reads Domain from the Container. Returns value with empty name
+// if domain is not specified.
+func ReadDomain(cnr Container) (res Domain) {
+	name := cnr.Attribute(container.SysAttributeName)
+	if name != "" {
+		res.SetName(name)
+		res.SetZone(cnr.Attribute(container.SysAttributeZone))
 	}
 
-	c.v2.SetPlacementPolicy(m)
+	return
 }
 
-// Marshal marshals Container into a protobuf binary form.
-func (c *Container) Marshal() ([]byte, error) {
-	return c.v2.StableMarshal(nil), nil
+// CalculateSignature calculates signature of the Container using provided signer
+// and writes it into dst. Signature instance MUST NOT be nil. CalculateSignature
+// is expected to be called after all the Container data is filled and before
+// saving the Container in the NeoFS network. Note that мany subsequent change
+// will most likely break the signature.
+//
+// See also VerifySignature.
+func CalculateSignature(dst *neofscrypto.Signature, cnr Container, signer ecdsa.PrivateKey) error {
+	return dst.Calculate(neofsecdsa.SignerRFC6979(signer), cnr.Marshal())
 }
 
-// Unmarshal unmarshals protobuf binary representation of Container.
-func (c *Container) Unmarshal(data []byte) error {
-	return c.v2.Unmarshal(data)
+// VerifySignature verifies Container signature calculated using CalculateSignature.
+// Result means signature correctness.
+func VerifySignature(sig neofscrypto.Signature, cnr Container) bool {
+	return sig.Verify(cnr.Marshal())
 }
 
-// MarshalJSON encodes Container to protobuf JSON format.
-func (c *Container) MarshalJSON() ([]byte, error) {
-	return c.v2.MarshalJSON()
+// CalculateIDFromBinary calculates identifier of the binary-encoded container
+// in CAS of the NeoFS containers and writes it into dst. ID instance MUST NOT
```

Comment:
one more CAS

Reply:
I'll keep it in sync with the discussion above.